client/components/videos-ui/index.jsx

```diff
@@ -3,42 +3,37 @@ import { Button, Gridicon } from '@automattic/components';
 import classNames from 'classnames';
 import { useTranslate } from 'i18n-calypso';
 import moment from 'moment';
-import { useEffect, useState, useMemo } from 'react';
+import { useEffect, useState } from 'react';
 import Notice from 'calypso/components/notice';
-import useCourseQuery from 'calypso/data/courses/use-course-query';
-import useUpdateUserCourseProgressionMutation from 'calypso/data/courses/use-update-user-course-progression-mutation';
+import {
+	COURSE_SLUGS,
+	useCourseData,
+	useUpdateUserCourseProgressionMutation,
+} from 'calypso/data/courses';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import VideoPlayer from './video-player';
 import './style.scss';
 
-const VideosUi = ( { HeaderBar, FooterBar, areVideosTranslated = true } ) => {
+const VideosUi = ( {
```

Comment:
Why are we introducing changes in this component? I had the idea that this component would only be used from this PR. I think changing this file adds an extra layer of complexity to this PR.

Reply:
I've created another PR to make the change especially for adding `useCourseData` hook. See: https://github.com/Automattic/wp-calypso/pull/59514